mlflow/server/handlers.py

```diff
@@ -1323,6 +1323,34 @@ def _delete_registered_model_tag():
     return _wrap_response(DeleteRegisteredModelTag.Response())
 
 
+def _validate_non_local_source_contains_relative_paths(source: str):
+    """
+    Validation check to ensure that sources that are provided that conform to the schemes:
+    http, https, or mlflow-artifacts do not contain relative path designations that are intended
+    to access local file system paths on the tracking server.
+
+    Example paths that this validation function is intended to find and raise an Exception if
+    passed:
+    "mlflow-artifacts://host:port/../../../../"
+    "http://host:port/api/2.0/mlflow-artifacts/artifacts/../../../../"
+    "https://host:port/api/2.0/mlflow-artifacts/artifacts/../../../../"
+    "/models/artifacts/../../../"
+    "s3:/my_bucket/models/path/../../other/path"
+    "file://path/to/../../../../some/where/you/should/not/be"
+    """
+    source_path = urllib.parse.urlparse(source).path
+    raw_source_path = pathlib.Path(source_path)
+    resolved_path = raw_source_path.resolve()
+
+    if resolved_path != raw_source_path:
```

Comment:
It looks like this doesn't work on Windows...

Reply:
looks like `WindowsPath().resolve()` prepends the pwd drive designator. Updated the logic and validated it works on my Windows PC.